### PR TITLE
Writing 3 Image-Related Guides

### PR DESCRIPTION
**Pull request by abidlabs**
This PR will include 3 new, related Guides:

- Image classification in PyTorch
- Image classification in Keras
- Image classification with Vision Transformers (using HF Hub)

### CODE REVIEW

File: guides/image_classification_in_tensorflow.md [hunk 1]
```diff
@@ -0,0 +1,88 @@
+# Image Classification in TensorFlow and Keras
+
+related_spaces: abidlabs/tensorflow-image-classifier
```
Comment (aliabd) [inline guides/image_classification_in_tensorflow.md:3]: This space doesn't exist. also iframing a 400 error below

please also include the full link when you correct it, as well as more related spaces like:
* https://huggingface.co/spaces/keras-io/keras-image-classifier
* https://huggingface.co/spaces/keras-io/Image-classification-with-EANet

File: guides/image_classification_in_tensorflow.md [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+tags: VISION, MOBILENET, TENSORFLOW
+
+## Introduction
+
+Image classification is a central task in computer vision. Building better classifiers to classify what object is present in a picture is an active area of research, as it has applications stretching from traffic control systems to satellite imaging. 
+
+Such models are perfect to use with Gradio's *image* input component, so in this tutorial we will build a web demo to classify images using Gradio. We will be able to build the whole web application in Python, and will look like this (try one of the examples!):
```
Comment (aliabd) [inline guides/image_classification_in_tensorflow.md:10]: > and will look like this

and it* will look like this

File: guides/image_classification_in_tensorflow.md [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+<iframe src="https://hf.space/gradioiframe/abidlabs/tensorflow-image-classifier/+" frameBorder="0" height="450" title="Gradio app" class="container p-0 flex-grow space-iframe" allow="accelerometer; ambient-light-sensor; autoplay; battery; camera; document-domain; encrypted-media; fullscreen; geolocation; gyroscope; layout-animations; legacy-image-formats; magnetometer; microphone; midi; oversized-images; payment; picture-in-picture; publickey-credentials-get; sync-xhr; usb; vr ; wake-lock; xr-spatial-tracking" sandbox="allow-forms allow-modals allow-popups allow-popups-to-escape-sandbox allow-same-origin allow-scripts allow-downloads"></iframe>
```
Comment (aliabd) [inline guides/image_classification_in_tensorflow.md:12]: space doesn't exist, but when fixed please modify height to fit without scrollbars. Try modifying in inspect element

File: guides/image_classification_in_tensorflow.md [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+Make sure you have the `gradio` Python package already [installed](/getting_started). We will be using a pretrained Keras image classification model, so you should also have `tensorflow` installed.
+
+## Step 1 — Setting up the Image Classification Model
+
+First, you will need a image classification model. For this tutorial, we will use a pretrained Mobile Net model, as it is easily downloadable from [Keras](https://keras.io/api/applications/mobilenet/). You can use a different pretrained model or train your own. 
+
+```python
+import tensorflow as tf
+
+inception_net = tf.keras.applications.MobileNetV2()
```
Comment (aliabd) [inline guides/image_classification_in_tensorflow.md:28]: does it make sense to call this variable inception_net and not mobile_net. obv not a big deal but will mess with search in the guides page if people are specifically looking for inception

File: guides/image_classification_in_tensorflow.md [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+
+def classify_image(inp):
+  inp = inp.reshape((-1, 224, 224, 3))
+  inp = tf.keras.applications.mobilenet_v2.preprocess_input(inp)
+  prediction = inception_net.predict(inp).flatten()
+  confidences = {labels[i]: float(prediction[i]) for i in range(1000)}
+  return confidences
+```
+
+Let's break this down. The function takes one parameters:
```
Comment (aliabd) [inline guides/image_classification_in_tensorflow.md:54]: > parameters

parameter*

File: guides/image_classification_in_tensorflow.md [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+
+* `confidences`: the predictions, as a dictionary whose keys are class labels and whose values are confidence probabilities
+
+## Step 3 — Creating a Gradio Interface
+
+Now that we have our predictive function set up, we can create a Gradio Interface around it. 
+
+In this case, the input component is a drag-and-drop image component. To create this input, we can use the `"gradio.inputs.Image"` class, which creates the component and handles the preprocessing to convert that to a numpy array. We will instantiate the class with a parameter that automatically preprocesses the input image to be 224 pixels by 224 pixels, which is the size that MobileNet expects.
+
+The output component will be a `"label"`, which displays the top labels in a nice form. Since we don't want to show all 1,000 class labels, we will customize it to show only the top 3 images.
```
Comment (aliabd) [inline guides/image_classification_in_tensorflow.md:68]: same comment as before on referencing a string shortcut that is not used

File: guides/image_classification_in_tensorflow.md [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+```python
+import gradio as gr
+
+gr.Interface(fn=classify_image, 
+             inputs=gr.inputs.Image(shape=(224, 224)),
+             outputs=gr.outputs.Label(num_top_classes=3),
+             examples=["banana.jpg", "car.jpg"]).launch()
+```
+
+This produces the following interface, which you can try right here in your browser (try uploading your own examples!):
```
Comment (aliabd) [inline guides/image_classification_in_tensorflow.md:81]: same comment as before 
> try uploading your own examples!

your own images*

File: guides/image_classification_in_tensorflow.md [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+
+<iframe src="https://hf.space/gradioiframe/abidlabs/tensorflow-image-classifier/+" frameBorder="0" height="450" title="Gradio app" class="container p-0 flex-grow space-iframe" allow="accelerometer; ambient-light-sensor; autoplay; battery; camera; document-domain; encrypted-media; fullscreen; geolocation; gyroscope; layout-animations; legacy-image-formats; magnetometer; microphone; midi; oversized-images; payment; picture-in-picture; publickey-credentials-get; sync-xhr; usb; vr ; wake-lock; xr-spatial-tracking" sandbox="allow-forms allow-modals allow-popups allow-popups-to-escape-sandbox allow-same-origin allow-scripts allow-downloads"></iframe>
```
Comment (aliabd) [inline guides/image_classification_in_tensorflow.md:83]: same comment on height

File: guides/image_classification_with_vision_transformers.md [hunk 9]
```diff
@@ -0,0 +1,55 @@
+# Image Classification with Vision Transformers
+
+related_spaces: abidlabs/vision-transformer
```
Comment (aliabd) [inline guides/image_classification_with_vision_transformers.md:3]: include full link and more similar spaces. Can be those using transformers not necessarily vision.

File: guides/image_classification_in_tensorflow.md [hunk 10]
```diff
@@ -0,0 +1,88 @@
+# Image Classification in TensorFlow and Keras
+
+related_spaces: abidlabs/tensorflow-image-classifier
+tags: VISION, MOBILENET, TENSORFLOW
```
Comment (aliabd) [inline guides/image_classification_in_tensorflow.md:4]: include IMAGE and LABEL tags

File: guides/image_classification_in_pytorch.md [hunk 11]
```diff
@@ -0,0 +1,91 @@
+# Image Classification in PyTorch
+
+related_spaces: abidlabs/pytorch-image-classifier
+tags: VISION, RESNET, PYTORCH
```
Comment (aliabd) [inline guides/image_classification_in_pytorch.md:4]: include IMAGE and LABEL tags

File: guides/image_classification_with_vision_transformers.md [hunk 12]
```diff
@@ -0,0 +1,55 @@
...
+tags: VISION, TRANSFORMERS, HUB
```
Comment (aliabd) [inline guides/image_classification_with_vision_transformers.md:4]: include LOAD, IMAGE and LABEL tags

File: guides/image_classification_with_vision_transformers.md [hunk 13]
```diff
@@ -0,0 +1,55 @@
...
+
+State-of-the-art image classifiers are based on the *transformers* architectures, originally popularized for NLP tasks. Such architectures are typically called vision transformers (ViT). Such models are perfect to use with Gradio's *image* input component, so in this tutorial we will build a web demo to classify images using Gradio. We will be able to build the whole web application in a **single line of Python**, and will look like this (try one of the examples!):
+
+<iframe src="https://hf.space/gradioiframe/abidlabs/vision-transformer/+" frameBorder="0" height="450" title="Gradio app" class="container p-0 flex-grow space-iframe" allow="accelerometer; ambient-light-sensor; autoplay; battery; camera; document-domain; encrypted-media; fullscreen; geolocation; gyroscope; layout-animations; legacy-image-formats; magnetometer; microphone; midi; oversized-images; payment; picture-in-picture; publickey-credentials-get; sync-xhr; usb; vr ; wake-lock; xr-spatial-tracking" sandbox="allow-forms allow-modals allow-popups allow-popups-to-escape-sandbox allow-same-origin allow-scripts allow-downloads"></iframe>
```
Comment (aliabd) [inline guides/image_classification_with_vision_transformers.md:12]: increase height to 660

File: guides/image_classification_with_vision_transformers.md [hunk 14]
```diff
@@ -0,0 +1,55 @@
+# Image Classification with Vision Transformers
+
+related_spaces: abidlabs/vision-transformer
+tags: VISION, TRANSFORMERS, HUB
+
+## Introduction
+
+Image classification is a central task in computer vision. Building better classifiers to classify what object is present in a picture is an active area of research, as it has applications stretching from facial recognition to manufacturing quality control. 
+
+State-of-the-art image classifiers are based on the *transformers* architectures, originally popularized for NLP tasks. Such architectures are typically called vision transformers (ViT). Such models are perfect to use with Gradio's *image* input component, so in this tutorial we will build a web demo to classify images using Gradio. We will be able to build the whole web application in a **single line of Python**, and will look like this (try one of the examples!):
```
Comment (aliabd) [inline guides/image_classification_with_vision_transformers.md:10]: > and will look like this

and it* will

File: guides/image_classification_with_vision_transformers.md [hunk 15]
```diff
@@ -0,0 +1,55 @@
...
+
+Let's get started!
+
+### Prerequisites
+
+Make sure you have the `gradio` Python package already [installed](/getting_started).
+
+## Step 1 — Choosing a Vision Image Classification Model
+
+First, you will need a image classification model. For this tutorial, we will use a model from the [Hugging Face Model Hub](https://huggingface.co/models). The Hub contains thousands of models covering dozens of different machine learning tasks.
```
Comment (aliabd) [inline guides/image_classification_with_vision_transformers.md:23]: > you will need a image classification

 need an* image

File: guides/image_classification_with_vision_transformers.md [hunk 16]
```diff
@@ -0,0 +1,55 @@
...
+
+### Prerequisites
+
+Make sure you have the `gradio` Python package already [installed](/getting_started).
+
+## Step 1 — Choosing a Vision Image Classification Model
+
+First, you will need a image classification model. For this tutorial, we will use a model from the [Hugging Face Model Hub](https://huggingface.co/models). The Hub contains thousands of models covering dozens of different machine learning tasks. 
+
+Expand the Tasks category on the left sidebar and select "Image Classification" as our task of interest. You will then see all of the models on the Hub that are designed to classify images.
```
Comment (aliabd) [inline guides/image_classification_with_vision_transformers.md:25]: you can also link directly to the models with image classification selected, if you want: https://huggingface.co/models?pipeline_tag=image-classification

File: guides/image_classification_with_vision_transformers.md [hunk 17]
```diff
@@ -0,0 +1,55 @@
...
+
+At the time of writing, the most popular one is `google/vit-base-patch16-224`, which has been trained on ImageNet images at a resolution of 224x224 pixels. We will use this model for our demo. 
+
+## Step 2 — Loading the Vision Transformer Model with Gradio
+
+When using a model from the Hugging Face Hub, you do not need to define the input or output components for the demo. Similarly, we do not need to be concerned with the details of preprocessing or postprocessing.
```
Comment (aliabd) [inline guides/image_classification_with_vision_transformers.md:31]: >  *you* do not need to ... Similarly, *we* do not

change both to either you or we

File: guides/image_classification_with_vision_transformers.md [hunk 18]
```diff
@@ -0,0 +1,55 @@
...
+             examples=["alligator.jpg", "laptop.jpg"]).launch()
+```
+
+Notice that we have added one more parameter, the `examples`, which allows us to prepopulate our interfaces with a few predefined examples. 
+
+This produces the following interface, which you can try right here in your browser. When you input an image, it is automatically preprocessed and sent to the Hugging Face Hub API, where it is passed through the model and returned as a human-interpretable prediction.  Try uploading your own image!
+
+<iframe src="https://hf.space/gradioiframe/abidlabs/vision-transformer/+" frameBorder="0" height="450" title="Gradio app" class="container p-0 flex-grow space-iframe" allow="accelerometer; ambient-light-sensor; autoplay; battery; camera; document-domain; encrypted-media; fullscreen; geolocation; gyroscope; layout-animations; legacy-image-formats; magnetometer; microphone; midi; oversized-images; payment; picture-in-picture; publickey-credentials-get; sync-xhr; usb; vr ; wake-lock; xr-spatial-tracking" sandbox="allow-forms allow-modals allow-popups allow-popups-to-escape-sandbox allow-same-origin allow-scripts allow-downloads"></iframe>
```
Comment (aliabd) [inline guides/image_classification_with_vision_transformers.md:50]: increase height to 660

File: guides/image_classification_with_vision_transformers.md [hunk 19]
```diff
@@ -0,0 +1,55 @@
...
+
+----------
+
+And you're done! In one line of code, you have built a web demo for an image classifier. If you'd like to share with others, try setting `share=True` when you `launch()` the Interface!
```
Comment (aliabd) [inline guides/image_classification_with_vision_transformers.md:54]: If you'd like to expand this guide to include more on `.load` (loading from spaces, from outside HF) it might be a good idea. We could also write a separate guide dedicated to that

File: guides/image_classification_in_pytorch.md [hunk 20]
```diff
@@ -0,0 +1,91 @@
+# Image Classification in PyTorch
+
+related_spaces: abidlabs/pytorch-image-classifier
```
[the remark this answers is not in the document]
Comment (abidlabs) [reply guides/image_classification_in_pytorch.md:3]: Great thanks

File: guides/image_classification_in_pytorch.md [hunk 21]
```diff
@@ -0,0 +1,91 @@
...
+    prediction = torch.nn.functional.softmax(model(inp)[0], dim=0)
+    confidences = {labels[i]: float(prediction[i]) for i in range(1000)}    
+  return confidences
+```
+
+Let's break this down. The function takes one parameters:
+
+* `inp`: the input image as a `numpy` array
+
+Then, the function converts the image to a PIL Image and then eventually a PyTorch `tensor`, passes it through the model, and returns:
```
[the remark this answers is not in the document]
Comment (abidlabs) [reply guides/image_classification_in_pytorch.md:61]: Good point

**Comment by abidlabs** [comment]
Awesome, thank you so much for the detailed review @aliabd!
